Modules/Loadable/Markups/MRML/vtkCurveMeasurementsCalculator.cxx

```diff
@@ -314,10 +329,180 @@ bool vtkCurveMeasurementsCalculator::CalculatePolyDataCurvature(vtkPolyData* pol
   return true;
 }
 
+//------------------------------------------------------------------------------
+bool vtkCurveMeasurementsCalculator::CalculatePolyDataTorsion(vtkPolyData* polyData)
+{
+  if (polyData == nullptr)
+    {
+    return false;
+    }
+  if (polyData->GetNumberOfPoints() == 0 || polyData->GetNumberOfLines() == 0)
+    {
+    return false;
+    }
+  if (this->InputMarkupsMRMLNode == nullptr)
+    {
+    vtkErrorMacro("Failed to find input markups node");
+    return false;
+    }
+
+  // Note: This algorithm has been implemented based on the information at https://mathworld.wolfram.com/Torsion.html
+  //       and based on the implementation of CalculatePolyDataCurvature
+
+  // Get binormals and tangents arrays
+  vtkParallelTransportFrame* curveCoordinateSystemGenerator = this->InputMarkupsMRMLNode->GetCurveCoordinateSystemGeneratorWorld();
+  if (!curveCoordinateSystemGenerator)
+    {
+    vtkErrorMacro("Failed to access coordinate system generator");
+    return false;
+    }
+  vtkPointData* pointData = polyData->GetPointData();
+  vtkDoubleArray* tangents = vtkDoubleArray::SafeDownCast(
+    pointData->GetAbstractArray(curveCoordinateSystemGenerator->GetTangentsArrayName()));
+  vtkDoubleArray* binormals = vtkDoubleArray::SafeDownCast(
+    pointData->GetAbstractArray(curveCoordinateSystemGenerator->GetBinormalsArrayName()));
+  // Sanity checks
+  vtkPolyData* curvePoly = curveCoordinateSystemGenerator->GetOutput();
+  if (curvePoly->GetNumberOfPoints() != polyData->GetNumberOfPoints())
+    {
+    // Sanity check
+    vtkErrorMacro("Number of points does not match between given poly data and the coordinate system generator poly data ("
+      << curvePoly->GetNumberOfPoints() << " != " <<  polyData->GetNumberOfPoints() << ")");
+    return false;
+    }
+  if (binormals->GetNumberOfTuples() != pointData->GetNumberOfTuples())
+    {
+    vtkErrorMacro("Number of data points does not match between coordinate system generator ("
+      << binormals->GetNumberOfTuples() << ") and poly data point data (" << pointData->GetNumberOfTuples() << ")");
+    return false;
+    }
+
+  vtkCellArray* lines = polyData->GetLines();
+  vtkNew<vtkIdList> linePoints;
+
+  lines->GetCell(0, linePoints);
+  vtkIdType numberOfPoints = // Last point in closed curve line is the first point
+    (this->CurveIsClosed ? linePoints->GetNumberOfIds()-1 : linePoints->GetNumberOfIds());
+  if (numberOfPoints < 3)
+    {
+    vtkErrorMacro("Number of curve points too low for torsion calculation (" << numberOfPoints << "), at least 3 needed");
+    return false;
+    }
+
+  // Initialize torsion array
+  vtkNew<vtkDoubleArray> torsionArray;
+  torsionArray->Initialize();
+  torsionArray->SetName("Torsion");
```

Comment:
This string should be accessible on the vtkCurveMeasurementsCalculator. Preferably it should be possible to set the array name, but at least it there should be a get method for getting the name.